src/dataflow/src/render/map_filter_project.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Renders a filter expression that may reference `mz_logical_timestamp`.
+//!
+//! There are restricted options for how one can reference this term in
+//! a maintained dataflow. Specifically, all predicates need to be of the
+//! form
+//! ```ignore
+//! mz_logical_timestamp cmp_op expr
+//! ```
+//! where `cmp_op` is a comparison operator (e.g. <, >, =, >=, or <=) and
+//! `expr` is an expression that does not contain `mz_logical_timestamp`.
+
+use differential_dataflow::lattice::Lattice;
+use differential_dataflow::AsCollection;
+use differential_dataflow::Collection;
+use timely::dataflow::Scope;
+use timely::progress::{timestamp::Refines, Timestamp};
+
+use dataflow_types::*;
+use expr::{MapFilterProject, MirRelationExpr};
+use repr::{Row, RowArena};
+
+use crate::operator::StreamExt;
+use crate::render::context::Context;
+use crate::render::datum_vec::DatumVec;
+
+impl<G, T> Context<G, MirRelationExpr, Row, T>
+where
+    G: Scope<Timestamp = repr::Timestamp>,
+    G::Timestamp: Lattice + Refines<T>,
+    T: Timestamp + Lattice,
+{
+    /// Renders a filter expression that may reference `mz_logical_timestamp`.
+    ///
+    /// There are restricted options for how one can reference this term in
+    /// a maintained dataflow. Specifically, all predicates need to be of the
+    /// form
+    /// ```ignore
+    /// mz_logical_timestamp cmp_op expr
+    /// ```
+    /// where `cmp_op` is a comparison operator (e.g. <, >, =, >=, or <=) and
+    /// `expr` is an expression that does not contain `mz_logical_timestamp`.
+    pub fn render_mfp_after(
```

Comment:
I was immediately confused by this name. after ... what?

Reply:
The argument to the method! Maybe it seems better in my mind, but
```
context.render_mfp_after(mfp, relation_expr);
```
I'm up for other words, but the other options were "atop" and .. idk I pick bad names for relating things. :)

Reply:
ah that works for me